### ISSUE TITLE
MER2024这篇论文半监督学习部分的问题

### ISSUE THREAD

**Issue by KKK92112** [ZebangCheng/Emotion-LLaMA#80]
作者您好，我是一名研0的学生，自己的话是对于情感分析这块兴趣较大，近期阅读了您所发表的针对mer2023和mer2024这两个比赛的论文，真的是收获颇多，有关2023这篇论文的工作我也已成功复现，真的对我的帮助很大。关于MER2024这篇的工作，我还是有一个关于打伪标签和自训练的问题想请教一下您：

1、您是只对2万个没打标签的样本打了伪标签吗，还是说您是排除了测试集的样本之后对10万多个样本打了伪标签之后再来训的？我看官方发布的公告说是有10万多个没打标签的样本，然后为了降低难度，最后将评估范围限制在了2万个样本。针对这个概念我一直有点模糊，这个比赛最后是要提交2万个样本的预测结果吗，还是说限制参赛者只能利用2万个没有标注的数据来提升自己模型的效果？

期待能够收到作者的解答：）

**Comment by KKK92112** [comment]
作者您好！对于您在"SZTU-CMU at MER2024: Improving Emotion-LLaMA with
Conv-Attention for Multimodal Emotion Recognition"这篇论文中所用到方法，有几个问题想请教一下您：
1、您在论文中所提到的卷积注意力模块，我在您的代码中也已找到了相应的部分，但我发现您似乎并没有用到卷积分支中的四个卷积块，想请教一下作者这部分的内容您是写在了别的地方吗？或者说是我哪里忽略了吗？
`class Attention(nn.Module):
    def __init__(self, output_dim, layers='256,128', dropout=0.3):
        super(Attention, self).__init__()

        dims = [1024, 1024, 1024, 4096, 4096, 4096, 4096]
        self.features_count = 7

        self.feats_prep = nn.ModuleList()
        for i in range(len(dims)):
            self.feats_prep.append(self.MLP(dims[i], layers, dropout))

        layers_list = list(map(lambda x: int(x), layers.split(',')))
        hiddendim = layers_list[-1] * self.features_count
        self.attention_mlp = self.MLP(hiddendim, layers, dropout)

        self.fc_att = nn.Linear(layers_list[-1], self.features_count)
        self.fc_out = nn.Linear(layers_list[-1], output_dim)
        self.softmax = nn.Softmax(dim=1)

    def MLP(self, input_dim, layers, dropout):
        all_layers = []
        layers = list(map(lambda x: int(x), layers.split(',')))
        for i in range(0, len(layers)):
            all_layers.append(nn.Linear(input_dim, layers[i]))
            all_layers.append(nn.ReLU())
            all_layers.append(nn.Dropout(dropout))
            input_dim = layers[i]
        module = nn.Sequential(*all_layers)
        return module

    def forward(self, feats):   # [1, 4, 4096]

        feats_hidden = []
        for i in range(self.features_count):
            feats_hidden.append(self.feats_prep[i](feats[i]))  # [batch_size, 128]

        # import pdb
        # pdb.set_trace()

        multi_hidden1 = torch.cat(feats_hidden, dim=1)  # [batch_size, 128*features_count]
        multi_hidden2 = torch.stack(feats_hidden, dim=2)  # [batch_size, 128, features_count]

        # multi_hidden1 = torch.reshape(x, (x.size(0), -1))  # [batch_size, 128*features_count] [1, 512]
        # multi_hidden2 = torch.transpose(x, 1, 2)  # [batch_size, 128, features_count] [1, 128, 4]


        attention = self.attention_mlp(multi_hidden1)
        attention = self.fc_att(attention)
        attention = torch.unsqueeze(attention, 2)  # [batch_size, features_count, 1]

        fused_feat = torch.matmul(multi_hidden2, attention)
        fused_feat = fused_feat.squeeze(dim=2)  # [batch_size, 128]
        # fused_feat = fused_feat.squeeze()  # [batch_size, 128]
        emos_out = self.fc_out(fused_feat)
        emos_pred = self.softmax(emos_out)
        return emos_pred`
2、对于您之前所提出的Emotion-LLaMA，您在MER2024这篇论文中是用的哪个.pth文件打的伪标签呢，是Emotion_LLaMA.pth还是checkpoint_best.pth？
3、对于特征提取这部分的内容您提取了视觉特征、音频特征和文本特征，针对视觉特征提取部分的内容我看您所提供的特征提取代码中似乎只有VideoMAE和MAE部分的代码，但是您在论文中提到所使用的视觉编码器总共有4个，针对MER2024这个数据集，我看您所公布的特征提取代码中也有关于MER2024提取的记录，请问您所公开的特征文件是按照您在MER2023这篇论文中所提到的方式进行提取的吗？还是说您是按照MER2024这篇论文所提到的方式进行提取的？
4、针对MER2024这篇论文的工作，请问您方便说明一下在训练时您是怎么做的5折交叉验证吗？您方便公布这部分训练的代码吗？

**Comment by ZebangCheng** [comment]
> 作者您好，我是一名研0的学生，自己的话是对于情感分析这块兴趣较大，近期阅读了您所发表的针对mer2023和mer2024这两个比赛的论文，真的是收获颇多，有关2023这篇论文的工作我也已成功复现，真的对我的帮助很大。关于MER2024这篇的工作，我还是有一个关于打伪标签和自训练的问题想请教一下您：
> 
> 1、您是只对2万个没打标签的样本打了伪标签吗，还是说您是排除了测试集的样本之后对10万多个样本打了伪标签之后再来训的？我看官方发布的公告说是有10万多个没打标签的样本，然后为了降低难度，最后将评估范围限制在了2万个样本。针对这个概念我一直有点模糊，这个比赛最后是要提交2万个样本的预测结果吗，还是说限制参赛者只能利用2万个没有标注的数据来提升自己模型的效果？
> 
> 期待能够收到作者的解答：）

1. `是只对2万个没打标签的样本打了伪标签吗，还是说您是排除了测试集的样本之后对10万多个样本打了伪标签之后再来训的？` 是的，我们只对2万个样本打了伪标签。由于是比赛期间的工作，但是我们并不知道测试集的样本，所以没有排除测试集的样本。
2. `这个比赛最后是要提交2万个样本的预测结果吗，还是说限制参赛者只能利用2万个没有标注的数据来提升自己模型的效果？` 这两万个样本是官方提供的list，包含测试集的样本。这个比赛参赛者可以使用全部大约12万个没有标注的视频来提升自己的模型的效果。但是在比赛期间，为了比赛公平，要提交2万个样本的预测结果。现在主办方已经开源了MER2024的测试集，所以在测试时只用预测测试集的样本就行。
